src/nvim/eval.c

```diff
@@ -4239,11 +4239,17 @@ static int eval7(
         // use its contents.
         s = deref_func_name((const char *)s, &len, &partial, !evaluate);
 
+        // Need to make a copy, in case evaluating the arguments makes
+        // the name invalid.
+        s = vim_strsave(s);
```

Comment:
Do we need the whole string or only `len` of it (`s = xmemdupz(s,len);`)? Looks like `s` can point to long strings:
```
len: 26 strlen(s): 260 s: vimfiler#get_status_string() : &ft == 'unite' ? unite#get_status_string() : &ft == 'vimshell' ? vimshell#get_status_string() : ('' != MyReadonly() ? MyReadonly() . ' ' : '') . ('' != fname ? fname : '[No Name]') . ('' != MyModified() ? ' ' . MyModified() : '')
```